tests/Microsoft.Bot.Builder.Dialogs.Tests/OAuthPromptTests.cs

```diff
@@ -693,51 +785,32 @@ private async Task PromptTimeoutEndsDialogTest(IActivity oauthPromptActivity)
                 Assert.Single(((Activity)activity).Attachments);
                 Assert.Equal(OAuthCard.ContentType, ((Activity)activity).Attachments[0].ContentType);
 
-                // Add a magic code to the adapter
-                adapter.AddUserToken(connectionName, activity.ChannelId, activity.Recipient.Id, token, magicCode);
+                Assert.Equal(InputHints.AcceptingInput, ((Activity)activity).InputHint);
 
-                // Add an exchangable token to the adapter
-                adapter.AddExchangeableToken(connectionName, activity.ChannelId, activity.Recipient.Id, exchangeToken, token);
+                // Prepare an EventActivity with a TokenResponse and send it to the botCallbackHandler
+                var eventActivity = CreateEventResponse(adapter, activity, ConnectionName, Token);
+                var ctx = new TurnContext(adapter, (Activity)eventActivity);
+                botCallbackHandler(ctx, CancellationToken.None);
             })
-            .Send(oauthPromptActivity)
-            .AssertReply("ended")
+            .AssertReply("Logged in.")
             .StartTestAsync();
         }
 
-        private Activity CreateEventResponse(TestAdapter adapter, IActivity activity, string connectionName, string token)
-        {
-            // add the token to the TestAdapter
-            adapter.AddUserToken(connectionName, activity.ChannelId, activity.Recipient.Id, token);
-
-            // send an event TokenResponse activity to the botCallback handler
-            var eventActivity = ((Activity)activity).CreateReply();
-            eventActivity.Type = ActivityTypes.Event;
-            var from = eventActivity.From;
-            eventActivity.From = eventActivity.Recipient;
-            eventActivity.Recipient = from;
-            eventActivity.Name = SignInConstants.TokenResponseEventName;
-            eventActivity.Value = JObject.FromObject(new TokenResponse()
-            {
-                ConnectionName = connectionName,
-                Token = token,
-            });
-
-            return eventActivity;
-        }
-
-        private async Task OAuthPromptEndOnInvalidMessageSetting()
+        private async Task PromptTimeoutEndsDialogTest(IActivity oauthPromptActivity)
         {
             var convoState = new ConversationState(new MemoryStorage());
             var dialogState = convoState.CreateProperty<DialogState>("dialogState");
 
             var adapter = new TestAdapter()
                 .Use(new AutoSaveStateMiddleware(convoState));
 
-            var connectionName = "myConnection";
+            const string magicCode = "888999";
```

Comment:
I see this 'magicCode' many times. Should we move it out like we did with 'exchangeToken'?

Reply:
done 👍